coredocument/coredocument.proto

```diff
@@ -167,3 +172,35 @@ message NFT {
   bytes registry_id = 1;
   bytes token_id = 2;
 }
+
+// Field represents a custom attribute
+message Field {
+  bytes key_hashed = 1;
+  bytes key_unhashed = 2;
+  FieldType type = 3;
+  oneof value {
+    string str_val = 4;
+    bytes byte_val = 5;
+    google.protobuf.Timestamp time_val = 6;
+    Signed sig_val = 7;
+  }
+}
+
+enum FieldType {
+  FIELD_TYPE_INVALID = 0;
+  FIELD_TYPE_INTEGER = 1;
+  FIELD_TYPE_DECIMAL = 2;
+  FIELD_TYPE_STRING = 3;
+  FIELD_TYPE_BYTES = 4;
+  FIELD_TYPE_TIMESTAMP = 5;
+  FIELD_TYPE_SIGNED = 6;
+}
+
+message Signed {
+  // doc id and version is required so that someone wouldn't be able to fake a signature on another document by copying the signed structure into that document.
```

Comment:
The signer centrifuge id is also needed, right?

Reply:
The doc_id and doc_version are redundant. We only need doc_version

Reply:
Together with the `doc_id` you could instantly verify if everything is correct.
Otherwise, you need to fetch the document with the `doc_version` from the db and compare the `doc_id`

Reply:
@manuelpolzhofer the document_root is in the same version. At the point where you're verifying the signature, it doesn't need to be in it.

Reply:
Let say we have version 7 I am adding the signature we have version 8. We have a couple of updates afterwards and you would like to verify the signature at version 13?

Without the doc_id in it, I still would need to fetch the version 7 out of the db. Otherwise, I could just add the signature from another invoice.